### PR TITLE
building: add Rust SDK documentation

### PR DESCRIPTION
**Pull request by DemesneGH**
Add documentation for developing OP-TEE applications in Rust.

Signed-off-by: Yuan Zhuang <zhuangyuan04@baidu.com>
Signed-off-by: Rong Fan <fanrong03@baidu.com>

### CODE REVIEW

**Comment by DemesneGH** [comment]
Related PR: 
https://github.com/OP-TEE/build/pull/521
https://github.com/OP-TEE/manifest/pull/200

File: building/developing_optee_applications_in_rust.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
+.. _developing_optee_applications_in_rust:
+
+###################################### 
+Developing OP-TEE Applications in Rust
```
Comment (jbech-linaro) [inline building/developing_optee_applications_in_rust.rst:4]: Tempted to suggest simply
`Rust Trusted Applications`
or
`Trusted Applications - Rust`

This will void wrapping the text on the menubar to the left.

Comment (DemesneGH) [reply building/developing_optee_applications_in_rust.rst:4]: How about simplifying the title to `Teaclave TrustZone SDK`? @mssun

Comment (mssun) [reply building/developing_optee_applications_in_rust.rst:4]: > `Rust Trusted Applications`
> or
> `Trusted Applications - Rust`

Hi, I'm okay with these two. But we also provide Rust crate (i.e., `optee_teec`) to help developing host apps in Rust. I don't know how to have a simple title to express the both efforts.

Comment (jbech-linaro) [reply building/developing_optee_applications_in_rust.rst:4]: How about `OP-TEE with Rust`?

File: building/developing_optee_applications_in_rust.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    Run QEMU:
+
+    .. code-block:: bash
+
+       $ (cd build && make run-only)
+
+To differentiate from `optee_examples
+<https://optee.readthedocs.io/en/latest/building/gits/optee_examples/optee_examples.html#>`_
+, all OP-TEE Rust examples are suffixed with ``-rs`` and can be found in
```
Comment (jforissier) [inline building/developing_optee_applications_in_rust.rst:72]: There is another difference: they don't have the `optee_example_` prefix. For example `/usr/bin/optee_example_hello_world` vs. `/usr/bin/hello_world-rs`.

File: building/developing_optee_applications_in_rust.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+*****************
+
+Currently, Teaclave TrustZone SDK is compatible with QEMUv8 (aarch64).
+
+Before building examples written with Teaclave TrustZone SDK, you should 
+clone the OP-TEE repo first. For QEMUv8, run:
+
+.. code-block:: bash
+
+    $ repo init -u https://github.com/OP-TEE/manifest.git -m qemu_v8.xml
```
Comment (jforissier) [inline building/developing_optee_applications_in_rust.rst:21]: Create a top-level directory first:
```
$ mkdir YOUR_OPTEE_DIR
$ cd YOUR_OPTEE_DIR
```

File: building/developing_optee_applications_in_rust.rst (hunk 1 above)
Comment (etienne-lms) [reply building/developing_optee_applications_in_rust.rst:4]: `Trusted and Client Applications with Rust`?

File: building/developing_optee_applications_in_rust.rst [hunk 4]
```diff
@@ -0,0 +1,97 @@
+.. _developing_optee_applications_in_rust:
+
+###################################### 
+Developing OP-TEE Applications in Rust
+######################################
+
+This document describes how to build OP-TEE examples written in `Rust
```
Comment (etienne-lms) [inline building/developing_optee_applications_in_rust.rst:7]: "OP-TEE client and trusted application examples written in Rust"

File: building/developing_optee_applications_in_rust.rst [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+    $ repo init -u https://github.com/OP-TEE/manifest.git -m qemu_v8.xml 
+    $ repo sync
+
+The source code of Teaclave TrustZone SDK is located in 
+``YOUR_OPTEE_DIR/optee_rust/`` containing a set of examples written in Rust
+using the SDK.
+
+For more information about building OP-TEE using QEMU, see `run OP-TEE using
+QEMU <https://optee.readthedocs.io/en/latest/building/devices/qemu.html>`_ .
```
Comment (etienne-lms) [inline building/developing_optee_applications_in_rust.rst:29]: https://optee.readthedocs.io/en/latest/building/devices/qemu.html#qemu-v8 ?

File: building/developing_optee_applications_in_rust.rst (hunk 1 above)
Comment (mssun) [reply building/developing_optee_applications_in_rust.rst:4]: I prefer `OP-TEE with Rust`. I believe this can be rendered in the menubar without wrapping.

Comment (DemesneGH) [reply building/developing_optee_applications_in_rust.rst:4]: Thanks all,
I'll update the title to `OP-TEE with Rust`.

File: building/optee_with_rust.rst [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+    .. code-block:: bash
+
+       $ (cd build && make run-only)
+
+To differentiate from `optee_examples
+<https://optee.readthedocs.io/en/latest/building/gits/optee_examples/optee_examples.html#>`_
+, OP-TEE Rust examples are not prefixed with `optee_example_` but suffixed with
+``-rs``. All of them are located in separate directories in
+``YOUR_OPTEE_DIR/optee_rust/examples``. More description about Rust examples can
```
Comment (etienne-lms) [inline building/optee_with_rust.rst:76]: I find the references a bit confusing between source filetrees and generated/installed binary file.
I would remove the sentence "All of them are located in separate directories in
``YOUR_OPTEE_DIR/optee_rust/examples``" since this path is already mentioned above.

File: building/optee_with_rust.rst [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+       $ ./build/soc_term.py 54321
+
+    Run QEMU:
+
+    .. code-block:: bash
+
+       $ (cd build && make run-only)
+
+To differentiate from `optee_examples
```
Comment (etienne-lms) [inline building/optee_with_rust.rst:72]:
```patch
-To differentiate from `optee_examples (ref)`_, OP-TEE Rust examples...
+To differentiate from client applications generated by `optee_examples (ref)`_, OP-TEE Rust examples...
```

File: building/optee_with_rust.rst [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+using the SDK.
+
+For more information about building OP-TEE using QEMUv8, see `run OP-TEE using
+QEMU
+<https://optee.readthedocs.io/en/latest/building/devices/qemu.html#qemu-v8>`_ .
+
+Compile Rust examples 
+*********************
+
+To build Rust examples in ``optee_rust/examples/`` and install them in
```
Comment (etienne-lms) [inline building/optee_with_rust.rst:36]: I find the sentence confusing: building *in* optee_rust/examples/? Rather build ~~*in*~~ optee_rust/examples/. I would rephrase to something like:
```
Rust examples applications are located in ``optee_rust/examples/``. To build and install them with Buildroot, run:

File: building/optee_with_rust.rst [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+To differentiate from `optee_examples
+<https://optee.readthedocs.io/en/latest/building/gits/optee_examples/optee_examples.html#>`_
+, OP-TEE Rust examples are not prefixed with `optee_example_` but suffixed with
+``-rs``. All of them are located in separate directories in
+``YOUR_OPTEE_DIR/optee_rust/examples``. More description about Rust examples can
+be found in `Overview of OP-TEE Rust Examples
+<https://teaclave.apache.org/trustzone-sdk-docs/overview-of-optee-rust-examples/>`_
+.
+
+In the compilation process, host applications are installed to ``/usr/bin/`` and
```
Comment (etienne-lms) [inline building/optee_with_rust.rst:81]: s/In the compilation/During the build process/

File: building/optee_with_rust.rst (hunk 6 above)
Comment (DemesneGH) [reply building/optee_with_rust.rst]: Updated. Thanks!

**Comment by etienne-lms** [comment]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Comment by DemesneGH** [comment]
Updated `$ (cd build && make OPTEE_RUST_ENABLE=y CFG_TEE_RAM_VA_SIZE=0x00300000)` 
to `$ (cd build && make toolchains && make OPTEE_RUST_ENABLE=y CFG_TEE_RAM_VA_SIZE=0x00300000)`
